lib/index.js

```diff
@@ -337,31 +337,55 @@ function Postgres(a, b) {
   }
 
   function listen(channel, fn) {
+    const listener = getListener();
+
     if (channel in listeners) {
       listeners[channel].push(fn)
-      return Promise.resolve(channel)
+      return Promise.resolve(listener.result)
```

Comment:
This will result in any listener receiving an unlisten function that only works for the initial listener.

Here's one of the test cases modified to test this.
```js
ot('multiple listeners and unlisten one', async() => {
  const listener = postgres(options)
      , xs = []

  await listener.listen('test', x => xs.push('1', x))
  const s2 = await listener.listen('test', x => xs.push('2', x))
  await listener.notify('test', 'a')
  await delay(50)
  await s2.unlisten()
  await listener.notify('test', 'b')
  await delay(50)
  listener.end();

  return ['1a2a1b', xs.join('')]
})
```

Reply:
Ah, right... unlisten is different for every `fn`

Reply:
fixed